File: src-tauri/src/database/models/provider.rs
use diesel::prelude::*;

#[derive(Queryable, Selectable)]
#[diesel(table_name = crate::schema::posts)]
#[diesel(check_for_backend(diesel::sqlite::Sqlite))]
pub struct Privder {
    pub id: i32,
    pub name: String,
    pub api: String,
    pub username: String,
    pub password: String,
    pub ip: String,
    pub port: i32,
    pub connection_type: i32
}

File: src/types/provider.ts
import { ProviderType } from "./providerType";

export interface Provider {
    id:              number;
    name:            string;
    api:             string;
    username:        string;
    password:        string;
    ip:              string;
    port:            number;
    connection_type: ProviderType;
}
